Nos/Controller/PagedNoteDataSource.swift

```diff
@@ -87,9 +87,14 @@ class PagedNoteDataSource<Header: View, EmptyPlaceholder: View>: NSObject, UICol
     // MARK: - UICollectionViewDataSource
     
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        let numberOfItems = fetchedResultsController.fetchedObjects?.count ?? 0
-        Log.debug("Number of items: \(numberOfItems) in section: \(section)")
-        return numberOfItems
+        let numberOfFetchedObjects = fetchedResultsController.fetchedObjects?.count ?? 0
+        // because we batch updates together to reduce animations but this function is called in between batches we
+        // need to account for the number of items queued for insertion or deletion. FetchedResultsController sees them
+        // but the collectionView doesn't yet.
+        let numberOfItemsInView = numberOfFetchedObjects - insertedIndexes.count + deletedIndexes.count
+        Log.debug("Number of items: \(numberOfFetchedObjects) in fetchedResultsController")
```

Comment:
Oh yeah, I intentionally left these in because I'm not sure the crash is fixed, and if it happens again I want the data. I'm thinking we can remove them a week or two after public deployment if we don't see the crash.